homeassistant/components/media_player/sonos.py

```diff
@@ -61,6 +61,10 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     global DEVICES
 
     if discovery_info:
+        # if device allready maped by config
+        if DEVICES:
```

Comment:
Done. I've test the old solution with my 8 device and it have work with autodiscovery. But you are right, it is not a good solution. The new one is better :+1: